http-server-undertow/src/test/groovy/io/micronaut/servlet/undertow/UndertowErrorSpec.groovy

```diff
@@ -35,14 +37,49 @@ class UndertowErrorSpec extends Specification {
         result.header(HttpHeaders.CONTENT_TYPE).startsWith(MediaType.TEXT_PLAIN)
     }
 
+    void "error that occurs with streaming response results in client receiving an error"() {
+        when:
+        client.toBlocking().exchange("/errors/stream-immediate", String)
+
+        then:
+        HttpClientResponseException ex = thrown(HttpClientResponseException)
+        ex.status == HttpStatus.INTERNAL_SERVER_ERROR
+        ex.response.body.orElseThrow() == "Internal Server Error: Immediate error"
+    }
+
+    void "error that occurs with streaming response after data sent results in client receiving incomplete data"() {
+        when:
+        client.toBlocking().exchange("/errors/stream-delayed", Integer[].class)
+
+        then:
+        HttpClientResponseException ex = thrown(HttpClientResponseException)
+        ex.status == HttpStatus.OK
```

Comment:
is 200 code expected for an error?

Reply:
@sdelamo Yes, because this is an odd case. As the error occurs during a chunked response, the status code has already been written and can't be changed. The only thing the server can do is just stop writing the data and end the response. The expected behavior being verified is that incomplete data is sent and thus the client exception happens when actually trying to deserialize the response.